pymongo_auth_aws/auth.py

```diff
@@ -103,17 +147,24 @@ def _aws_temp_credentials():
                 'temporary MONGODB-AWS credentials could not be obtained, '
                 'error: %s' % (exc,))
 
+    # See https://docs.aws.amazon.com/cli/latest/reference/sts/assume-role.html#examples for expected result format.
     try:
         temp_user = res_json['AccessKeyId']
         temp_password = res_json['SecretAccessKey']
-        token = res_json['Token']
+        if 'SessionToken' in res_json:
+            session_token = res_json['SessionToken']
+        else:
+            session_token = res_json['Token']
```

Comment:
Is this SessionToken/Token change related to caching? Only "Token" is documented in the spec: https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#ecs-endpoint

Reply:
The [Credentials](https://docs.aws.amazon.com/STS/latest/APIReference/API_Credentials.html) object returned by an STS call has a `SessionToken`.  I'll make sure to document this in the new spec.

Reply:
Ah, I see what happened, the two calls give different fields.  I updated the logic and the tests.